### PR TITLE
Updates Fortran documentation for error codes

### PR DESCRIPTION
**Pull request by ashao**
The examples in the Fortran documentation examples used a mixture of subroutines and functions. The former was deprecated when error handling was implemented to allow the clients to bubble up a return code that Fortran applications can check. This has now been updated to allow others to use the documents as an accurate jumping off point.

Closes https://github.com/CrayLabs/SmartRedis/issues/335

### CODE REVIEW

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [inline doc/sr_fortran_walkthrough.rst:81]: Please add the line to check the return code as well

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [inline doc/sr_fortran_walkthrough.rst:33]: Please don't refer to enum members by their values -- it limits our ability to change them in the future. Document instead that SRNoError is the "all is OK" value.

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [inline doc/sr_fortran_walkthrough.rst:27]: Capitalize the 'c' in C++ (this applies below as well)

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [inline doc/sr_fortran_walkthrough.rst:211]: show checking the error code

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [inline doc/sr_fortran_walkthrough.rst:238]: show checking the error code

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [inline doc/sr_fortran_walkthrough.rst:326]: show checking the error code

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [inline doc/sr_fortran_walkthrough.rst:335]: show checking the error code

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [inline doc/sr_fortran_walkthrough.rst:366]: show checking the error code

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [inline doc/sr_fortran_walkthrough.rst:374]: show checking the error code

File: doc/sr_integration.rst
Comment (billschereriii) [inline doc/sr_integration.rst:47]: yy = no no

File: doc/sr_integration.rst
Comment (billschereriii) [inline doc/sr_integration.rst:180]: delete extra space?

File: doc/sr_integration.rst
Comment (billschereriii) [inline doc/sr_integration.rst:196]: is there a reason to prefer /= vs .ne.?

**Review by billschereriii — CHANGES_REQUESTED** [review]
Great content, just a few minor quibbles. Also, please update the changelog

File: doc/sr_integration.rst
Comment (ashao) [reply doc/sr_integration.rst]: Not really; they are synonyms with each other. I changed these to `.ne.` for consistency

File: doc/sr_fortran_walkthrough.rst
Comment (billschereriii) [reply doc/sr_fortran_walkthrough.rst:27]: Yes it is. Looke LC to me initially for some reason

File: doc/sr_integration.rst
Comment (billschereriii) [inline doc/sr_integration.rst:205]: one more /= here

File: doc/sr_integration.rst
Comment (billschereriii) [inline doc/sr_integration.rst:220]: /=

File: doc/sr_integration.rst
Comment (billschereriii) [inline doc/sr_integration.rst:226]: /=

**Comment by ashao** [comment]
I'm going to wait until #285 is merged so we get a clean pass on the CI.

**Review by billschereriii — APPROVED** [review]
Thanks for taking care of these updates! This will help our users getting started with SmartRedis in Fortran quite a bit

**Pull request merged**
